src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,109 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private T[] elementData;
+
+    public ArrayList() {
+        this.elementData = (T[]) new Object[DEFAULT_CAPACITY];
+        this.size = 0;
+    }
+
+    public void indexValidator(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Index is not valid");
+        }
+    }
+
+    public Object[] grow(T[] elementData, int size) {
```

Comment:
Method grow should be void, because it changes the main array, e.g. this.elementData. 
So, you can directly access this  `this.elementData`.
```suggestion
    public void grow(T[] elementData, int size) {
```